File: backend/src/main/java/springweb/backend/model/ItemDto.java
package springweb.backend.model;

import lombok.Builder;
import lombok.With;

import java.util.List;

@With
@Builder
public record ItemDto (
        String name,
        String img,
        String description,
        ItemCategory category,
        ItemStatus status,
        List<Double> geocode,
        String owner
) {
}

File: frontend/src/models/ItemDto.ts
import {ItemCategory} from "./ItemCategory.ts";
import {ItemStatus} from "./ItemStatus.ts";

export type ItemDto = {
    name: string,
    img: string,
    description: string,
    category: ItemCategory,
    status: ItemStatus,
    geocode: [number,number],
    owner: string
}